src/ui/pages/SidePage/components/IncomingRequest/IncomingRequest.tsx

```diff
@@ -1,54 +1,40 @@
 import { useEffect, useState, useMemo } from "react";
 import "./IncomingRequest.scss";
-import { Agent } from "../../../../../core/agent/agent";
 import { useAppDispatch, useAppSelector } from "../../../../../store/hooks";
 import {
-  dequeueCredentialRequest,
+  dequeueIncomingRequest,
   getQueueIncomingRequest,
 } from "../../../../../store/reducers/stateCache";
 import { SidePageContentProps } from "../../SidePage.types";
-import { RequestComponent } from "./components/RequestComponent";
-import {
-  IncomingRequestProps,
-  IncomingRequestType,
-} from "../../../../../store/reducers/stateCache/stateCache.types";
-import { getConnectedWallet } from "../../../../../store/reducers/walletConnectionsCache";
+import { SignRequest } from "./components/SignRequest"; // Import SignRequest component
+import { PeerConnectSigningEventRequest } from "../../../../../store/reducers/stateCache/stateCache.types";
 
 const IncomingRequest = ({ open, setOpenPage }: SidePageContentProps) => {
   const pageId = "incoming-request";
   const dispatch = useAppDispatch();
   const queueIncomingRequest = useAppSelector(getQueueIncomingRequest);
-  const connectedWallet = useAppSelector(getConnectedWallet);
   const incomingRequest = useMemo(() => {
     if (
       !queueIncomingRequest.isProcessing ||
       !queueIncomingRequest.queues.length
     ) {
       return;
     } else {
-      return queueIncomingRequest.queues[0];
+      return queueIncomingRequest.queues[0] as PeerConnectSigningEventRequest;
     }
   }, [queueIncomingRequest]);
   const [initiateAnimation, setInitiateAnimation] = useState(false);
-  const [requestData, setRequestData] = useState<IncomingRequestProps>();
+  const [requestData, setRequestData] =
+    useState<PeerConnectSigningEventRequest>();
   const ANIMATION_DELAY = 4000;
   const [blur, setBlur] = useState(false);
 
   useEffect(() => {
     if (!incomingRequest) {
       return;
     }
-    if (
-      incomingRequest.type === IncomingRequestType.PEER_CONNECT_SIGN &&
-      (!connectedWallet ||
-        connectedWallet.id !== incomingRequest.peerConnection?.id)
-    ) {
-      handleReset();
```

Comment:
This `handleReset` was added in case we have a request in the queue from a previously connected dApp (i.e. we switched dApp, or got disconnected) - this needs to stay